hera_qm/firstcal_metrics.py

```diff
@@ -312,12 +312,33 @@ def __init__(self, calfits_file):
         self.UVC = UVCal()
         self.UVC.read_calfits(calfits_file)
 
+        if len(self.UVC.jones_array) > 1:
+            raise ValueError('Sorry, only single pol firstcal solutions are '
+                             'currently supported.')
+        pol_dict = {-5: 'x', -6: 'y'}
+        try:
+            self.pol = pol_dict[self.UVC.jones_array[0]]
+        except KeyError:
+            raise ValueError('Sorry, only calibration polarizations "x" and '
+                             '"y" are currently supported.')
+
+        # Get the firstcal.rotated_antenna.metric file
+        if self.UVC.cal_type == 'gain':
+            with open(calfits_file + '.rotated_metric.json') as rot_file:
+                rmetric = json.load(rot_file)
+            self.delays = np.array([rmetric['delays'][str(ai)] for ai in self.UVC.ant_array]).squeeze()
+            self.rotated_antennas = { int(k): self.pol for k in rmetric['rotated_antennas'] }
```

Comment:
I think it would be more robust to *not* make `firstcal_metrics` dependent on a `.json` file in order to run if the `cal_type` is == `gain`, which seems to be the standard from here on out since `hera_cal.firstcal` is outputting solutions with gains not delays. One thing that is easy is to just make `firstcal_metrics` calculate the delays if given the gains, which is pretty straightforward. Here is some code to do that, which matches exactly (to float precision) of the `delays` in the `.json` file. @zakiali I can insert this into this line here, or if you don't want merge conflicts you could do it yourself. Let me know what you think.
```
    freqs = self.UVC.freq_array.squeeze()
    fc_gains = np.moveaxis(self.UVC.gain_array, 2, 3)[:, 0, :, :, :]
    d_nu = np.mean(freqs[1:]-freqs[:-1])
    d_phi = np.abs(np.mean(np.angle(fc_gains)[:, :, 1:, :] - np.angle(fc_gains)[:, :, :-1, :], axis=2))
    fc_delays = (d_phi / d_nu)/(2*np.pi)
```

Reply:
I like this idea, though I have a few concerns. 

First off, use np.diff rather than `freqs[1:]-freqs[:-1]`

Second, instead of mean, use median, which protects you from phase wraps. Alternatively, use unwrap. Or maybe do both?

Third, I think the abs gives you all positive delays...but some delays are negative!

Fourth, you also need to automatically detect the offsets in addition to the delays.

Finally, I think it's important that the delays and offsets are stored in an easily readable way somewhere...at the very least in the firstcal metrics json.

Reply:
@jsdillon yes you are right, thanks, here is updated block:
```
fc_gains = np.moveaxis(self.UVC.gain_array, 2, 3)[:, 0, :, :, 0]
fc_phi = np.unwrap(np.angle(fc_gains))
d_nu = np.mean(np.diff(freqs))
d_phi = np.mean(fc_phi[:, :, 1:] - fc_phi[:, :, :-1], axis=2)
fc_delays = (d_phi / d_nu)/(-2*np.pi)
```

Reply:
@jsdillon about the offsets: the json file that zaki writes out doesn't contain the offsets: is that a new capability you'd like `firstcal_metrics` to store, or are you referring to something else?

Reply:
Offsets is the same as the notion of rotated antennas. Rotated antennas have pi or -pi offsets, non-rotated antennas have 0 offset (e.g. the delay at freq=0). The firstcal_metrics needs to record rotated antennas, which are derivable from the gains, since the gains will +/- 1 * e^(-2 * pi * i * tau * nu) depending on whether or not the antenna is detected by firstcal as rotated.

Reply:
Also, I feel like the median rather than the mean is more robust if something goes wrong with the unwrapping.

Reply:
Yea, this sounds good to me. Feel free to merge it in to this PR.